contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -62,7 +74,121 @@ contract VehicleStream is AccessControlInternal {
+
+        emit VehicleStreamAssociated(vehicleId, streamId);
+
+        streamRegistry.grantPermission(
+            streamId,
+            StreamrConfiguratorStorage.getStorage().dimoStreamrNode,
+            IStreamRegistry.PermissionType.Publish
+        );
+    }
+
+    /**
+     * @notice Allows a vehicle owner to associate an existing stream Id from their vehicle Id
+     * @dev This contract must have Publish and Grant Streamr permissions
+     * @dev Reverts if vehicle id does not exist, caller is not the owner,
+     * stream Id does not exist, DIMO Registry does not have grant permission or DIMO Streamr Node the publish permission
+     * @param vehicleId Vehicle node Id
+     */
+    function setVehicleStream(
+        uint256 vehicleId,
+        string calldata streamId
+    ) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        StreamrConfiguratorStorage.Storage
+            storage scs = StreamrConfiguratorStorage.getStorage();
+        IStreamRegistry streamRegistry = IStreamRegistry(scs.streamRegistry);
+        VehicleStreamStorage.Storage storage vs = VehicleStreamStorage
+            .getStorage();
+
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        if (!streamRegistry.exists(streamId)) {
+            revert StreamDoesNotExist(streamId);
+        }
+
+        if (
+            !streamRegistry.hasPermission(
+                streamId,
+                scs.dimoStreamrNode,
+                IStreamRegistry.PermissionType.Publish
+            )
+        ) {
+            revert NoStreamrPermission(
+                scs.dimoStreamrNode,
+                IStreamRegistry.PermissionType.Publish
+            );
+        }
+        if (
+            !streamRegistry.hasPermission(
+                streamId,
+                address(this),
+                IStreamRegistry.PermissionType.Grant
+            )
+        ) {
+            revert NoStreamrPermission(
+                address(this),
+                IStreamRegistry.PermissionType.Grant
+            );
+        }
+
+        string memory oldStreamId = vs.streams[vehicleId];
+        if (bytes(oldStreamId).length != 0) {
+            emit VehicleStreamDissociated(vehicleId, oldStreamId);
+        }
+
+        vs.streams[vehicleId] = streamId;
+
+        emit VehicleStreamAssociated(vehicleId, streamId);
+    }
+
+    /**
+     * @notice Allows a vehicle owner to dissociate the stream Id from their vehicle Id
+     * @dev Reverts if vehicle id does not exist or caller is not the owner
+     * @param vehicleId Vehicle node Id
+     */
+    function dissociateVehicleStream(uint256 vehicleId) external {
```

Comment:
It's a bit odd to me to have
`dissociateVehicleStream` -> `VehicleStreamDissociated`
but
`setVehicleStream` -> `VehicleStreamAssociated`

Reply:
VehicleStreamUnset?

Reply:
Let's do set and unset. Convers more cases.